File: academic-system/src/main/java/org/example/service/ServicoSeguranca.java
package org.example.service;

import org.example.model.Usuario;
import org.example.repository.RepositorioUsuario;
import org.example.exception.ExcecaoAutenticacao;
import org.example.exception.ExcecaoAutorizacao;
import java.util.Optional;
import java.util.logging.Logger;

public class ServicoSeguranca {
    private static final Logger logger = Logger.getLogger(ServicoSeguranca.class.getName());
    private final RepositorioUsuario repositorio;

    public ServicoSeguranca(RepositorioUsuario repositorio) {
        this.repositorio = repositorio;
    }

    public Usuario autenticar(String username, String password) {
        Optional<Usuario> usuario = repositorio.buscarPorUsername(username);
        if (usuario.isEmpty() || !usuario.get().getPassword().equals(password)) {
            logger.warning("Falha de autenticação para o usuário: " + username);
            throw new ExcecaoAutenticacao("Credenciais inválidas.");
        }
        logger.info("Usuário autenticado com sucesso: " + username);
        return usuario.get();
    }

    public void autorizarAdmin(String role) {
        if (!"ADMIN".equals(role)) {
            logger.warning("Tentativa de acesso negada para o papel: " + role);
            throw new ExcecaoAutorizacao("Acesso negado: Requer privilégios de administrador.");
        }
        logger.info("Autorização de administrador concedida.");
    }

    public void logout(String username) {
        logger.info("Usuário efetuou logout com sucesso: " + username);
    }
}

File: academic-system-csharp/SistemaAcademico/src/SistemaAcademico/Services/ServicoSeguranca.cs
using SistemaAcademico.Models;
using SistemaAcademico.Repositories;
using SistemaAcademico.Exceptions;
using Microsoft.Extensions.Logging;

namespace SistemaAcademico.Services;

// Equivalente à classe Java: ServicoSeguranca
// java.util.Optional<Usuario> → Usuario? (nullable)
// Injeção de dependência via construtor (preservada)
public class ServicoSeguranca
{
    private static readonly ILogger<ServicoSeguranca> logger =
        LoggerFactory.Create(b => b.AddConsole()).CreateLogger<ServicoSeguranca>();

    private readonly IRepositorioUsuario repositorio;

    public ServicoSeguranca(IRepositorioUsuario repositorio)
    {
        this.repositorio = repositorio;
    }

    public Usuario Autenticar(string username, string password)
    {
        Usuario? usuario = repositorio.BuscarPorUsername(username);
        if (usuario == null || !usuario.Password.Equals(password))
        {
            logger.LogWarning("Falha de autenticação para o usuário: {Username}", username);
            throw new ExcecaoAutenticacao("Credenciais inválidas.");
        }
        logger.LogInformation("Usuário autenticado com sucesso: {Username}", username);
        return usuario;
    }

    public void AutorizarAdmin(string role)
    {
        if (!"ADMIN".Equals(role))
        {
            logger.LogWarning("Tentativa de acesso negada para o papel: {Role}", role);
            throw new ExcecaoAutorizacao("Acesso negado: Requer privilégios de administrador.");
        }
        logger.LogInformation("Autorização de administrador concedida.");
    }

    public void Logout(string username)
    {
        logger.LogInformation("Usuário efetuou logout com sucesso: {Username}", username);
    }
}
